doc/guidelines/e2e_frontend_coding_guidelines.md

```diff
@@ -0,0 +1,145 @@
+
+Locate element by `data-testid`.
+The most resilient attribute.
+
+#### [getByLabel](https://playwright.dev/docs/locators#locate-by-label)
+
+> Most form controls usually have dedicated labels that could be conveniently used to interact with the form. In this case, you can locate the control by its associated label using [page.getByLabel()](https://playwright.dev/docs/api/class-page#page-get-by-label).
+
+Allows locating input elements by the text of the associated <label> or aria-labelledby element, or by the aria-label attribute.
+This is a reasonable alternative popular for input forms.
+
+#### [getByText](https://playwright.dev/docs/locators#locate-by-text)
+
+> We recommend using text locators to find non interactive elements like div, span, p, etc. For interactive elements like button, a, input, etc. use [role locators](https://playwright.dev/docs/locators#locate-by-role).
+
+Allows locating elements that contain given text.
+This alternative should be picked only if the other three are not possible.
+It is not very resilient and can frequently break the tests.
+
+#### [Playwright suggested locators](https://playwright.dev/docs/input)
+
+- text inputs: **label**
+- checkboxes: **label**
+- radio buttons: **label**
+- select options: **label**
+- mouse clicks: **role or text**
+- type characters: **locator**
+- keys and shortcuts: **text or role** **<- this is the only place where text is fine**
+- upload files and focus elements: **label**
+
+### 2. User facing or resilience?
+
+When chosing the right locator it is important to ask yourself one question.
+Do we want to test what the user perceives?
+If so, go with `roles`.
+If user-centricity is not so important on this element use `data-testid`.
+In case of doubt use `data-testid`.
+
+### 3. Naming Conventions
+
+#### Prefix
+
+We will add `test-` as a prefix to every `data-testid` element.
+
+#### Unique identifiers
+
+Well chosen names.
+
+```html
+<!-- Good -->
+<input type="text" data-testid="test-username-input" />
+
+<!-- Bad -->
+<input type="text" data-testid="test-input-1" />
+```
+
+#### Human readable
+
+No abbreviations.
+
+```html
+<!-- Good -->
+<button data-testid="test-login-button">Login</button>
+
+<!-- Bad -->
+<button data-testid="test-lgnbtn">Login</button>
+```
+
+#### Accessibility Best Practices
+
+Utilize for.
+
+```html
+<!-- Good -->
+<label for="test-username-input">Username:</label>
+<input type="text" data-testid="test-username-input" />
```

Comment:
I think this should be the same as the i18n identifier for "Username:".
Use real code we have, then it probably makes more sense for you, e.g. in `frontend/src/features/seeds/components/CreateSeedForm.tsx`:

```
            <SimpleFormInput
              labelText={t('seeds:additional_name')}
              id="name"
              data-test-id="seeds:additional_name"
              ....
            />

```


Of course for cases where there is no text you somehow need to come up with other `data-testid`.